cassovary-core/src/main/scala/com/twitter/cassovary/util/SortedArrayOps.scala

```diff
@@ -0,0 +1,85 @@
+package com.twitter.cassovary.util
+
+import java.util
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+object SortedArrayOps {
+  /**
+   * Finds element in the SortedSeq using provided ordering.
+   *
+   * Assumes that the collection is sorted using this ordering.
+   */
+  def findUsingOrdering(array: Array[Int], t: Int): Boolean = {
+    util.Arrays.binarySearch(array, t) match {
+      case idx if idx < 0 => false
+      case idx => true
+    }
+  }
+
+  /**
+   * Linear intersection of two sorted arrays.
+   */
+  def intersectSorted(array: Array[Int], that: Array[Int]): Array[Int] = {
+    val result = Array.ofDim[Int](array.length + that.length)
+    var resultIdx = 0
+    var arrayIdx = 0
+    var thatIdx = 0
+
+    while (arrayIdx < array.length && thatIdx < that.length) {
+      if (array(arrayIdx) == that(thatIdx)) {
+        result(resultIdx) = array(arrayIdx)
+        resultIdx += 1
+        arrayIdx += 1
+        thatIdx += 1
+      } else if (array(arrayIdx) < that(thatIdx)) {
+        arrayIdx += 1
+      } else {
+        thatIdx += 1
+      }
+    }
+
+    util.Arrays.copyOfRange(result, 0, resultIdx)
+  }
+
+  /**
+   * Linear intersection of two sorted arrays.
+   */
+  def setsumSorted(array: Array[Int], that: Array[Int]): Array[Int] = {
+    val result = Array.ofDim[Int](array.length + that.length)
+    var resultIdx = 0
+    var arrayIdx = 0
+    var thatIdx = 0
+
+    def putInDest(x: Int): Unit = {
+      result(resultIdx) = x
+      resultIdx += 1
+    }
+
+    while (arrayIdx < array.length || thatIdx < that.length) {
+      if (arrayIdx == array.length) {
+        putInDest(that(thatIdx))
+        thatIdx += 1
+      } else if (thatIdx == array.length) {
```

Comment:
did you mean that.length here instead of array.length?

Also, please check that tests cover the conditions.